pyEpiabm/pyEpiabm/sweep/spatial_sweep.py

```diff
@@ -42,52 +44,102 @@ def __call__(self, time: float):
                                   if person.is_infectious()]
             infector = random.choice(possible_infectors)
 
-            # Each infection event corresponds to a infectee cell
-            # on the cell list
-            for infectee_cell in cell_list:
-
-                # Sample at random from the infectee cell to find
-                # an infectee
-                infectee = random.sample(infectee_cell.persons, 1)[0]
-                if not infectee.is_susceptible():
-                    continue
-
-                # force of infection specific to cells and people
-                # involved in the infection event
-                force_of_infection = SpatialInfection.\
-                    space_foi(cell, infectee_cell, infector, infectee)
-
-                # Compare a uniform random number to the force of
-                # infection to see whether an infection event
-                # occurs in this timestep between the given
-                # persons.
-                r = random.random()
-
-                if r < force_of_infection:
-                    infectee_cell.enqueue_person(infectee)
+            if Parameters.instance().do_CovidSim:
+                # Chooses cells based on a cumulative transmission array
+                # one and a time a tests each infection event.
+                while number_to_infect > 0:
+                    # Weighting for cell choice in Covidsim uses cum_trans and
+                    # invCDF arrays, but really can't see how these are
+                    # initialised. Have used the number of susceptible for now.
+                    weights = [cell2.compartment_counter.retrieve()
+                               [InfectionStatus.Susceptible]
+                               for cell2 in possible_infectee_cells]
+                    infectee_cell = random.choices(possible_infectee_cells,
+                                                   weights=weights, k=1)[0]
+                    # Sample at random from the infectee cell to find
+                    # an infectee
+                    infectee = random.sample(infectee_cell.persons, 1)[0]
+                    infection_distance = DistanceFunctions.dist(
+                     cell.location, infectee_cell.location) / Parameters.\
+                        instance().infection_radius
+                    if (infection_distance < random.random()):
+                        # Covidsim rejects the infection event if the distance
+                        # between infector/infectee is too large.
+                        self.do_infection_event(infector, infectee, timestep)
+                        number_to_infect -= 1
+
+            else:
+                # Chooses a list of cells (with replacement) for each infection
+                # event to occur in. Specifically inter-cell infections
+                # so can't be the same cell
+
+                distance_weights = [1/DistanceFunctions.dist(
```

Comment:
Could be worth handling the case where distance=0 to avoid divide by zero

Reply:
Implemented a max_weight if we get divide by zero. Just in the code for now, would suggest it as a global param, but feel I'm making a lot of those atm